lblod/ember-rdfa-editor
# GN-4583: Remove node properties/backlink pointing to removed RDFA node

### Overview

GN-4583: Remove node properties/backlink pointing to removed RDFA node
Also remove mentions of "alternative backspace", as it won't be "alternative" anymore 🤔 

##### connected issues and PRs:

[GN-4583](https://binnenland.atlassian.net/browse/GN-4583)

### Setup

1. Checkout

### How to test/reproduce

1. Use `RdfaTest` 
2. Select an RDFA node (e.g. Cmd + Click it)
3. Press backspace
4. Observe correct properties/backlinks were deleted
5. Observe selection was deleted/modified correctly
6. Select multiple RDFA nodes with text in between
2. Press backspace
3. Observe correct properties/backlinks were deleted
4. Observe selection was deleted/modified correctly
5. Select multiple RDFA nodes with text in between
2. Enter some text, e.g. "hello"
3. Observe correct properties/backlinks were deleted
4. Observe selection was deleted/modified correctly


### Challenges/uncertainties

2. I have tried to implement a visible selector for the user, so he can press it directly to select the node, but I've decided against it, as it was not working properly. It was depending on `handleDOMEvents` and using [posAtDOM](https://prosemirror.net/docs/ref/#view.EditorView.posAtDOM) detect what node to select, but in the end it was not the best implementation, as `posAtDOM` was not alway selecting what I hoped it would, even with the `bias` param adjustments. Better way here would be to implement an `EmberView` for RDFA nodes, then selecting a node will be a breeze. Users still can use "Cmd + Click".
3. 
   ```ts 
   isolating: true,
   selectable: true,
   ```
   is a free way to select node on backspace, cannot not like it :)  


### Checks PR readiness
- [X] npm lint - target branch is failing
